### DISCUSSION TITLE
Decomposition of F# grammar

### DISCUSSION THREAD

**Discussion by xp44mm** [fsharp/fslang-suggestions#1218] [General]
Although the F # source file is a large grammar file, the grammar of F# is relatively independent. References include:

- [F# spec](https://fsharp.org/specs/language-spec/)
- [F# documentation](https://learn.microsoft.com/en-us/dotnet/fsharp/)
- source code [pars.fsy](https://github.com/dotnet/fsharp/blob/main/src/Compiler/pars.fsy) and Tips from the F# compiler when you input f# source code into ide.

First, I extracted the grammar of TypeArgument, which is `typ` in pars.fsy

BNF file: [typeArgument.fsyacc](https://github.com/xp44mm/FslexFsyacc/blob/master/FSharpAnatomy/typeArgument.fsyacc)

test data:[TypeArgumentCompilerTest.fs](https://github.com/xp44mm/FslexFsyacc/blob/master/FSharpAnatomy.Test/TypeArgumentCompilerTest.fs)

I shared the decomposed grammar file, Welcome to discuss and study together.

**Comment by abelbraaksma** [comment]
Interesting, are you trying to create your own compiler of F#? You posted this under ‘F# Suggestions’, did you mean to make a language suggestion, or are you asking for a code review? Note that you can get the TAST from F#, which you could use for correctness checking of your own compiler.

**Comment by vzarytovskii** [comment]
I, once again, suggest having such discussions over at https://github.com/dotnet/fsharp/discussions/categories/q-a, since it usually have more people paying attention to it.
This repo is generally for **language suggestions** and related discussions, and generally have less people following it.

**Reply by abelbraaksma** [reply]
Agreed! We should have some way of moving these issues. Not everyone will always be aware of ‘what goes where’, happy to help with such a cleanup (if at all possible).

On the flip side, people ask for help, so we should help ;). I’ll make a mental note of linking to the q&a when next time answering such questions.

**Reply by vzarytovskii** [reply]
Yeah, people are still welcome to ask questions anywhere (including here), just saying, it'll be a better chance getting answer/discussion started in dotnet/fsharp or SO.
